akka-cluster-typed/src/main/scala/akka/cluster/ddata/typed/javadsl/ReplicatorMessageAdapter.scala

```diff
@@ -0,0 +1,164 @@
+    }
+  }
+
+  /**
+   * Send a [[Replicator.Update]] request to the replicator. The [[Replicator.UpdateResponse]]
+   * message is transformed to the message protocol of the requesting actor with the given
+   * `responseAdapter` function.
+   *
+   * Note that `createRequest` is a function that creates the `Update` message from the provided
+   * `ActorRef[UpdateResponse]` that the the replicator will send the response message back through.
+   * Use that `ActorRef[UpdateResponse]` as the `replyTo` parameter in the `Update` message.
+   */
+  def askUpdate(
+      createRequest: JFunction[ActorRef[Replicator.UpdateResponse[B]], Replicator.Update[B]],
+      responseAdapter: JFunction[Replicator.UpdateResponse[B], A]): Unit = {
+    context.asScala.ask[Replicator.Update[B], Replicator.UpdateResponse[B]](replicator)(askReplyTo =>
+      createRequest(askReplyTo)) {
+      case Success(value) => responseAdapter(value)
+      case Failure(ex)    => throw ex // unexpected ask timeout
+    }
+  }
+
+  /**
+   * Send a [[Replicator.Get]] request to the replicator. The [[Replicator.GetResponse]]
+   * message is transformed to the message protocol of the requesting actor with the given
+   * `responseAdapter` function.
+   *
+   * Note that `createRequest` is a function that creates the `Get` message from the provided
+   * `ActorRef[GetResponse]` that the the replicator will send the response message back through.
+   * Use that `ActorRef[GetResponse]` as the `replyTo` parameter in the `Get` message.
+   */
+  def askGet(
+      createRequest: JFunction[ActorRef[Replicator.GetResponse[B]], Replicator.Get[B]],
+      responseAdapter: JFunction[Replicator.GetResponse[B], A]): Unit = {
+    context.asScala.ask[Replicator.Get[B], Replicator.GetResponse[B]](replicator)(askReplyTo =>
+      createRequest(askReplyTo)) {
+      case Success(value) => responseAdapter(value)
+      case Failure(ex)    => throw ex // unexpected ask timeout
+    }
+  }
+
+  /**
+   * Send a [[Replicator.Delete]] request to the replicator. The [[Replicator.DeleteResponse]]
+   * message is transformed to the message protocol of the requesting actor with the given
+   * `responseAdapter` function.
+   *
+   * Note that `createRequest` is a function that creates the `Delete` message from the provided
+   * `ActorRef[DeleteResponse]` that the the replicator will send the response message back through.
+   * Use that `ActorRef[DeleteResponse]` as the `replyTo` parameter in the `Delete` message.
+   */
+  def askDelete(
+      createRequest: JFunction[ActorRef[Replicator.DeleteResponse[B]], Replicator.Delete[B]],
+      responseAdapter: JFunction[Replicator.DeleteResponse[B], A]): Unit = {
+    context.asScala.ask[Replicator.Delete[B], Replicator.DeleteResponse[B]](replicator)(askReplyTo =>
+      createRequest(askReplyTo)) {
+      case Success(value) => responseAdapter(value)
+      case Failure(ex)    => throw ex // unexpected ask timeout
+    }
```

Comment:
Funky formatting for all of these, maybe breaking the line different makes it less funky?